NGF: Update user architecture guide (#953)
nginx/documentation: content/ngf/overview/gateway-architecture.md

### Data Plane: Autonomous Traffic Management

Each NGINX data plane pod can be provisioned as an independent Deployment or DaemonSet containing an `nginx` container. This container runs both the `nginx` process and the [NGINX agent](https://github.com/nginx/agent), which is responsible for:

- Applying configurations: The agent receives updates from the control plane and applies them to the NGINX instance.
- Handling reloads: NGINX Agent handles configuration reconciliation and reloading NGINX, eliminating the need for shared volumes or Unix signals between the control plane and data plane pods.

### Gateway Resource Management

Users can have multiple gateways running side-by-side in the same cluster. This supports flexible operation and isolation across Gateways:

- Concurrent Gateways: Multiple Gateway objects can run simultaneously within a single installation.
- 1:1 resource mapping: Each Gateway resource corresponds uniquely to a dedicated data plane deployment, ensuring clear delineation of ownership and operational segregation.

---

> ## High-Level Overview of NGINX Gateway Fabric in Action

Reviewer: ```suggestion
## High-level overview of NGINX Gateway Fabric in execution
```

Sentence case

This figure depicts an example of NGINX Gateway Fabric exposing three web applications within a Kubernetes cluster to clients on the internet:

```mermaid

> graph LR

Reviewer 2: couple notes on this diagram:

Let's change the NGF Pod name to NGF Control Plane Pod, and NGINX Pod to NGINX Data Plane Pod, to reinforce the concepts that we already introduced in this doc.

Also, the NGF Pod is outside the Kubernetes cluster. It should be inside it, in the nginx-gateway namespace.

(same goes for the table that follows the diagram)

    style HTTPRouteAAndApplications fill:#E0FFFF,stroke:#333,stroke-width:2px
    style HTTPRouteBandApplications fill:#E0FFFF,stroke:#333,stroke-width:2px
    style UserDevA fill:#FFA07A,stroke:#333,stroke-width:2px
    style HTTPRouteA fill:#FFA07A,stroke:#333,stroke-width:2px
    style HTTPRouteB fill:#FFA07A,stroke:#333,stroke-width:2px
    style ApplicationA fill:#FFA07A,stroke:#333,stroke-width:2px
    style ApplicationB fill:#FFA07A,stroke:#333,stroke-width:2px
    style ClientA fill:#FFA07A,stroke:#333,stroke-width:2px
    style UserDevB fill:#87CEEB,stroke:#333,stroke-width:2px
    style HTTPRouteC fill:#87CEEB,stroke:#333,stroke-width:2px
    style ApplicationC fill:#87CEEB,stroke:#333,stroke-width:2px
    style ClientB fill:#87CEEB,stroke:#333,stroke-width:2px
    style PublicEndpoint fill:#FFD700,stroke:#333,stroke-width:2px
```

{{< call-out "note" >}} The figure does not show many of the necessary Kubernetes resources the Cluster Operators and Application Developers need to create, like deployment and services. {{< /call-out >}}

The figure shows:

{{< bootstrap-table "table table-bordered table-striped table-responsive" >}}

> | **Category**           | **Description**                                                                                                                                       |

Reviewer 2: We technically should not be using the `NGF` acronym in our docs, so let's replace that anywhere with NGINX Gateway Fabric.

| **Applications**        | - _Application A_: Deployed by Developer A (2 pods) and routed to by Gateway A.<br>- _Application B_: Deployed by Developer A (1 pod) and routed to by Gateway A.<br>- _Application C_: Deployed by Developer B (1 pod) and routed to by Gateway B. |
| **NGINX Pods**          | - _NGINX Pod A_: Handles traffic from Gateway A:<br>&nbsp;&nbsp;&nbsp;• _NGINX Process A_: Routes to Application A and Application B.<br>&nbsp;&nbsp;&nbsp;• _NGINX Agent A_: Updates configuration via gRPC.<br>- _NGINX Pod B_: Handles traffic from Gateway B:<br>&nbsp;&nbsp;&nbsp;• _NGINX Process B_: Routes to Application C.<br>&nbsp;&nbsp;&nbsp;• _NGINX Agent B_: Updates configuration via gRPC. |
| **Traffic Flow**        | - _Client A_:<br>&nbsp;&nbsp;&nbsp;1. Sends requests to `a.example.com` via Public Endpoint.<br>&nbsp;&nbsp;&nbsp;2. Routed to Application A by NGINX Process A.<br>- _Client B_:<br>&nbsp;&nbsp;&nbsp;1. Sends requests to `c.other-example.com` via Public Endpoint.<br>&nbsp;&nbsp;&nbsp;2. Routed to Application C by NGINX Process B. |

> | **Public Endpoint**     | A shared entry point (TCP Load Balancer or NodePort) that exposes services externally and forwards client traffic into the cluster.                   |

Reviewer 2: ```suggestion
| **Public Endpoint**     | A shared entry point (TCP Load Balancer or NodePort) that exposes the NGINX Service externally and forwards client traffic into the cluster.                   |
```

> graph LR

Reviewer 2: Similar comments for this diagram and table

| 7  | HTTP, HTTPS            | _Client → NGINX Worker_: Clients send traffic (e.g., HTTP/HTTPS requests) to the NGINX Worker. These are typically routed via a public LoadBalancer or NodePort to expose NGINX. |
{{% /bootstrap-table %}}

---

### NGINX Plus Features and Benefits

NGINX Gateway Fabric supports both NGINX Open Source and NGINX Plus. While the previous diagram shows NGINX Open Source, using NGINX Plus provides additional capabilities, including:

- The ability for administrators to connect to the NGINX Plus API on port 8765 (restricted to localhost by default).
- Dynamic updates to upstream servers without requiring a full reload:
  - Changes to upstream servers, such as application scaling (e.g., adding or removing pods in Kubernetes), can be applied using the [NGINX Plus API](http://nginx.org/en/docs/http/ngx_http_api_module.html).
  - This reduces the frequency of configuration reloads, minimizing potential disruptions and improving system stability during updates.

These features contribute to reduced downtime, improved performance during scaling events, and more fine-grained control over traffic management.

---

### Resilience and Fault Isolation

> This architecture separates the control plane and data plane, creating clear operational boundaries that improve resilience and fault isolation. It also enhances scalability, security, and reliability while reducing the risk of failures affecting both components.

Reviewer: "enhances scalability, security, and reliability" this is ambiguous marketing voice.

It should be explained properly, or the text should be restructured to excise it by referring to detail elsewhere on the page.

Author: I think i will go ahead and remove this line. Adding more information about scalability, reliability and security seems verbose. That information is widely available across the docs for NGF. I was rewording this and felt like it was unnecessary.

> #### Control Plane Resilience

Reviewer: ```suggestion
#### Control plane resilience
```
